js/huobi.js

```diff
@@ -5683,43 +5689,42 @@ module.exports = class huobi extends Exchange {
             [ marginMode, params ] = this.handleMarginModeAndParams ('fetchFundingHistory', params);
             marginMode = (marginMode === undefined) ? 'cross' : marginMode;
             if (marginMode === 'isolated') {
-                request['margin_account'] = market['id'];
+                request['mar_acct'] = market['id'];
             } else {
-                request['margin_account'] = market['quoteId'];
+                request['mar_acct'] = market['quoteId'];
             }
         } else {
             if (marketType === 'swap') {
-                method = 'contractPrivatePostSwapApiV1SwapFinancialRecordExact';
-                request['contract_code'] = market['id'];
+                method = 'contractPrivatePostSwapApiV3SwapFinancialRecordExact';
+                request['contract'] = market['id'];
+                //
+                //     {
+                //         "code": 200,
+                //         "msg": "",
+                //         "data": [
+                //             {
+                //                 "query_id": 138798248,
+                //                 "id": 117840,
+                //                 "type": 5,
+                //                 "amount": -0.024464850000000000,
+                //                 "ts": 1638758435635,
+                //                 "contract_code": "BTC-USDT-211210",
+                //                 "asset": "USDT",
+                //                 "margin_account": "USDT",
+                //                 "face_margin_account": ""
+                //             }
+                //         ],
+                //         "ts": 1604312615051
+                //     }
+                //
             } else {
-                throw new ExchangeError (this.id + ' fetchFundingHistory() only makes sense for swap contracts');
+                method = 'contractPrivatePostApiV3ContractFinancialRecordExact';
```

Comment:
Idk why there was exception before, and why it doesn't make sense to have funding history for `inverse futures` ?  it also has it's own endpoint, and the concept shouldnt be any different from `inverse swap`, as funding history applies to both of them.

Reply:
hum, so probably was a bug